packages/zksend/src/experimental/wallet/index.ts

```
@@ -0,0 +1,444 @@
+	}
+
+	get version() {
+		return '1.0.0' as const;
+	}
+
+	get chains() {
+		return [SUI_MAINNET_CHAIN] as const;
+	}
+
+	get accounts() {
+		return this.#accounts;
+	}
+
+	get features(): StandardConnectFeature &
+		StandardDisconnectFeature &
+		StandardEventsFeature &
+		SuiSignTransactionBlockFeature &
+		SuiSignTransactionFeature &
+		SuiSignPersonalMessageFeature {
+		return {
+			'standard:connect': {
+				version: '1.0.0',
+				connect: this.#connect,
+			},
+			'standard:disconnect': {
+				version: '1.0.0',
+				disconnect: this.#disconnect,
+			},
+			'standard:events': {
+				version: '1.0.0',
+				on: this.#on,
+			},
+			'sui:signTransactionBlock': {
+				version: '1.0.0',
+				signTransactionBlock: this.#signTransactionBlock,
+			},
+			'sui:signTransaction': {
+				version: '2.0.0',
+				signTransaction: this.#signTransaction,
+			},
+			'sui:signPersonalMessage': {
+				version: '1.0.0',
+				signPersonalMessage: this.#signPersonalMessage,
+			},
+		};
+	}
+
+	constructor({
+		name,
+		network,
+		address,
+		origin = DEFAULT_STASHED_ORIGIN,
+	}: {
+		name: string;
+		network: StashedSupportedNetwork;
+		origin?: string;
+		address?: string | null;
+	}) {
+		this.#accounts = [];
+		this.#events = mitt();
+		this.#origin = origin;
+		this.#name = name;
+		this.#network = network;
+
+		if (address) {
+			this.#setAccount(address);
+		}
+	}
+
+	#signTransactionBlock: SuiSignTransactionBlockMethod = async ({ transactionBlock, account }) => {
+		transactionBlock.setSenderIfNotSet(account.address);
+
+		const data = transactionBlock.serialize();
+
+		const popup = new StashedPopup({
+			name: this.#name,
+			origin: this.#origin,
+			network: this.#network,
```

ignoring `chain` here feels kinda problematic.   My preference would be for wallet implementations to be multi-network.  If we are going to tie the wallet to a specific network, we need to at least throw an error here when chain is set incorrectly

---

The signing flows on wallet support chain so we should be able to pass it here